src/entities/Subscription.ts

```diff
@@ -0,0 +1,69 @@
+import { SubscriptionSet } from './SubscriptionSet';
+import { SubscribeCapable, SubscriptionOptions } from './commonTypes';
+
+export class Subscription implements SubscribeCapable {
+  private channelNames: string[] = [];
+  private groupNames: string[] = [];
+  private options?: SubscriptionOptions;
+  private pubnub: any;
+  private eventEmitter: any;
+
+  constructor({
+    channels,
+    channelGroups,
+    subscriptionOptions,
+    eventEmitter,
+    pubnub,
+  }: {
+    channels: string[];
+    channelGroups: string[];
+    subscriptionOptions?: SubscriptionOptions;
+    eventEmitter: any;
+    pubnub: any;
+  }) {
+    this.channelNames = channels;
+    this.groupNames = channelGroups;
+    this.options = subscriptionOptions;
+    this.pubnub = pubnub;
+    this.eventEmitter = eventEmitter;
+  }
+
+  subscribe() {
+    this.pubnub.subscribe({
+      channels: this.channelNames,
+      channelGroups: this.groupNames,
+      ...(this.options?.cursor?.timetoken && { timetoken: this.options.cursor.timetoken }),
+    });
+  }
+  unsubscribe() {
+    this.pubnub.unsubscribe({ channels: this.channelNames, channelGroups: this.groupNames });
+  }
```

Comment:
As I understand we currently use simplified version without entities usage tracking?

Reply:
Listeners will pass entity names to subscribe/unsubscribe call as it is. 

to make this logic at one central place it's added in [this](https://github.com/pubnub/javascript/blob/CLEN-1785-listeners/src/event-engine/index.ts) file.  So that presence apis will be called /optimised accordingly.

* added utility methods to assort unique/overlapping channel/groups. to make decision whether to remove/add incoming channel/group names into subscription/presence event engine.